custom_components/tesla_custom/__init__.py

```diff
@@ -27,30 +40,49 @@
     DEFAULT_SCAN_INTERVAL,
     DEFAULT_WAKE_ON_START,
     DOMAIN,
+    ICONS,
     MIN_SCAN_INTERVAL,
     PLATFORMS,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_TOKEN): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): vol.All(cv.positive_int, vol.Clamp(min=MIN_SCAN_INTERVAL)),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
 
 @callback
-def _async_save_tokens(hass, config_entry, access_token, refresh_token, expiration):
```

Comment:
Let's keep the expiration in the config_flow. Otherwise we will automatically refresh the token in the first 5 minutes after start up. Given how paranoid Tesla is, we want to minimize auth attempts.